src/momento/_utilities/_data_validation.py

```diff
@@ -21,18 +22,18 @@ def _as_bytes(
     raise errors.InvalidArgumentError(error_message + str(type(data)))
 
 
-def _validate_ttl(ttl_seconds: int) -> None:
-    if not isinstance(ttl_seconds, int) or ttl_seconds < 0:
-        raise errors.InvalidArgumentError("TTL Seconds must be a non-negative integer")
+def _validate_ttl(ttl: timedelta) -> None:
+    if not isinstance(ttl, timedelta) or ttl.total_seconds() < 0:
+        raise errors.InvalidArgumentError("TTL timedelta must be a non-negative integer")
 
 
-def _validate_ttl_minutes(ttl_minutes: int) -> None:
-    if not isinstance(ttl_minutes, int) or ttl_minutes < 0:
-        raise errors.InvalidArgumentError("TTL Minutes must be a non-negative integer")
+def _validate_ttl_seconds(ttl_seconds: int) -> None:
+    if not isinstance(ttl_seconds, int) or ttl_seconds < 0:
+        raise errors.InvalidArgumentError("TTL Seconds must be a non-negative integer")
```

Comment:
I think this is only used internally because prepare_set_request takes seconds. If prepare_set_request is changed to use timedelta this can be deleted.

Reply:
Good observation. I pushed the conversion from timedelta to seconds into the `prepare_set_request()` methods.